### PR TITLE
Add catalan number benchmark

### PR DESCRIPTION
**Pull request by susan-garry**
Recursively compute the nth [Catalan number](https://en.wikipedia.org/wiki/Catalan_number).

### CODE REVIEW

File: benchmarks/catalan.bril [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  idx: int = id zero;
+  n: int = sub n one;
+.while:
+  guard1: bool = le idx n;
+  br guard1 .while.body .while.end;
+.while.body:
+  n2: int = sub n idx;
+  v1: int = call @catalan idx;
+  v2: int = call @catalan n2;
+  elti = mul v1 v2;
```
Comment (sampsyo) [inline benchmarks/catalan.bril:27]: Looks like this is missing a type annotation.

**Review by sampsyo — CHANGES_REQUESTED** [review]
Looks good overall! But brilck is complaining that there is a missing type annotation in the benchmark code.

File: benchmarks/catalan.bril (hunk 1 above)
Comment (susan-garry) [reply benchmarks/catalan.bril]: Should be fixed now!

**Review by sampsyo — APPROVED** [review]
Looks like it's all set!

**Pull request merged**